app/metal-controller-manager/cmd/agent/main.go

```diff
@@ -118,25 +146,65 @@ func main() {
 		log.Fatal(err)
 	}
 
-	log.Println("Reading SMBIOS")
-
-	s, err := smbios.New()
-	if err != nil {
-		log.Fatal(err)
+	var state *string
+	if state = procfs.ProcCmdline().Get("sidero.state").First(); state == nil {
+		log.Fatal(fmt.Errorf("sidero.state is required"))
 	}
 
 	var endpoint *string
 	if endpoint = procfs.ProcCmdline().Get("sidero.endpoint").First(); endpoint == nil {
 		log.Fatal(fmt.Errorf("no endpoint found"))
 	}
 
-	log.Printf("Creating resource via %q", *endpoint)
+	log.Println("Reading SMBIOS")
 
-	if err = create(*endpoint, s); err != nil {
+	s, err := smbios.New()
+	if err != nil {
 		log.Fatal(err)
 	}
 
-	log.Println("Discovery complete")
+	switch metalv1alpha1.State(*state) {
+	case metalv1alpha1.StateUnknown:
+		log.Printf("Creating resource via %q", *endpoint)
+
+		if err = create(*endpoint, s); err != nil {
```

Comment:
my only comment: should we create server in `decomissioned` state so that it goes through wipe before it is available?

Reply:
Sounds like it could be a good idea. @rsmitty Thoughts?

Reply:
What if instead of marking it as decommissioned, which would require a second boot, we just run the cleanup code before we register the server?

Reply:
that sounds good as well!

Reply:
I decided not to wipe on registration since I am afraid of mistakenly wiping servers that contain important data, and are mistakenly booted into this environment (maybe a misconfigured DHCP option?).

Reply:
I think there should be no risk on wiping if we implement it correctly with `Server` lookup: if `Server` object doesn't exist (no such UUID), or `Server` is marked for wiping, wipe it.

If Server is `InUse` and doesn't need wiping - don't wipe it ;)

Reply:
I am less worried about things intended to be in the domain of Sidero, and more worried about machines that have accidentally been PXE booted with the Sidero services.

Reply:
Yeah, it would be a pretty high price to pay for an easy-to-have-happen accident.  It seems like the simple existence check for the Server should be sufficient, though.  That's essentially an implicit "admission" flag.